### DISCUSSION TITLE
Python requirement for Carla 0.9.12?

### DISCUSSION THREAD

**Discussion by smitsavaliya-1258** [carla-simulator/carla#4519] [General]
which python version is required for Carla 0.9.12? I know 0.9.9 required python 3.7 but i need to know about 0.9.12 ?

**Comment by rrrushan** [comment]
in my case 3.8 works fine

**Comment by hitabm** [comment]
Take a look at your egg of whl files in "PythonAPI/carla/dist" directory. In my distro their names are:
carla-0.9.12-**py3.7**-win-amd64.egg & carla-0.9.12-**cp37-cp37m**-win_amd64.whl
This means they need python 3.7 to work. Other versions may work to but it's better to use the exact version to prevent later problems.
